linkerd/app/admin/src/server/json.rs

```diff
@@ -41,26 +44,28 @@ pub(crate) fn accepts_json<B>(req: &http::Request<B>) -> Result<(), http::Respon
             tracing::warn!(?accept, "Accept header will not accept 'application/json'");
             return Err(http::Response::builder()
                 .status(StatusCode::NOT_ACCEPTABLE)
-                .body(JSON_MIME.into())
+                .body(BoxBody::new::<String>(JSON_MIME.into()))
                 .expect("builder with known status code must not fail"));
         }
     }
 
     Ok(())
 }
 
-fn mk_rsp(status: StatusCode, val: &impl serde::Serialize) -> http::Response<Body> {
+fn mk_rsp(status: StatusCode, val: &impl serde::Serialize) -> http::Response<BoxBody> {
     match serde_json::to_vec(val) {
         Ok(json) => http::Response::builder()
             .status(status)
             .header(header::CONTENT_TYPE, JSON_HEADER_VAL.clone())
-            .body(json.into())
+            .body(BoxBody::new(http_body::Full::new(bytes::Bytes::from(json))))
```

Comment:
`BoxBody::new(http_body::Full::new(bytes::Bytes::from(` is even more cumbersome, is there a way to shorten this with another `From` impl?

Reply:
i wrestled with this one a bit, and some of the trouble feels like a fundamental reflection of how the hyper interface will change in 1.0. there used to be a https://docs.rs/hyper/0.14.31/hyper/body/struct.Body.html type included that let us use a simple `.into()` before.

as we prepare for upgrading, we'll need to move towards providing our own `B: Body` type. that'll be a https://docs.rs/http-body-util/latest/http_body_util/struct.Full.html in the 1.0 interface, which expects a `D: Buf` [buffer.](https://docs.rs/bytes/1.6.0/bytes/buf/trait.Buf.html)

so, some of this _is_ just a bit more cumbersome, but 4557952 tries to mitigate that a bit by hoisting this out of the `match` arm and leaning on `Result::map` to apply these constructors.

does that look a little better to you, @sfleen?

Reply:
Pulling that out helps a fair amount I think. It's unfortunate that the hyper API has boxed us in like that, but I don't see a way around it that isn't fraught with footguns.